src/main/java/dal/fatih/todoly/Task.java

```diff
@@ -1,19 +1,15 @@
 package dal.fatih.todoly;
 
 import java.util.Date;
-import java.util.concurrent.atomic.AtomicInteger;
 
 public class Task {
 
-    private String id;

```

Comment:
Keep id in the task, don't remove it. When you have a task object, you should be able to know what the id is, without needing the map. Even though using the task id as the key of the map is a correct thing, that is not for removing an information from your Pojo class ( Task ) here.